### ISSUE TITLE
Early stopping

### ISSUE THREAD

**Issue by cklyne** [applied-ml-bde/bde#34]
## Idea
Optional early stopping should be added to the estimator fitting and jitted_training methods.
It should accept:
 - A patience value.
 - A metric to track (should default to the loss function).

## Comments
Integrating it with the jitted training methods is probably non-trivial due to the limitations on jitted methods. 
My guess is that the method which scans over `jitted_training_epoch` should scan an intermediate method which evaluates the early stopping condition. Using `jax.lax.cond` if it is met it should call `jitted_training_epoch`, and if not it should call a new method with exactly the same signature which return the same params and nan values for the history.
Alternatively we could use `jax.lax.while`, but it seems to be less efficient than `jax.lax.scan`.
